lib/conversion.c

```diff
@@ -123,17 +124,22 @@ static bool tpm2_convert_pubkey_ssl(TPMT_PUBLIC *public, pubkey_format format, c
         goto error;
     }
 
-    ssl_rsa_key->e = BN_bin2bn((void*)&exponent, sizeof(exponent), NULL);
-    ssl_rsa_key->n = BN_bin2bn(
-            (public->unique).rsa.t.buffer,
-            (public->unique).rsa.t.size,
-            NULL);
+    e = BN_bin2bn((void*)&exponent, sizeof(exponent), NULL);
+    n = BN_bin2bn(public->unique.rsa.t.buffer, public->unique.rsa.t.size,
+                  NULL);
 
-    if (!ssl_rsa_key->n || !ssl_rsa_key->e) {
+    if (!n || !e) {
         print_ssl_error("Failed to convert data to SSL internal format");
         goto error;
     }
 
+#if OPENSSL_VERSION_NUMBER < 0x1010000fL /* OpenSSL 1.1.0 */
+    ssl_rsa_key->e = e;
+    ssl_rsa_key->n = n;
+#else
+    RSA_set0_key(ssl_rsa_key, n, e, NULL);
```

Comment:
`RSA_set0_key` returns 0 on failure. That should be handled.

In that error case it would also be needed to call `BN_free` on `e` and `n`, because ownership has not been transferred to the `ssl_rsa_key` structure.

Reply:
@mgerstner done in the next version.